lib/charms/postgresql_k8s/v1/postgresql.py

```diff
@@ -202,129 +220,243 @@ def create_access_groups(self) -> None:
             logger.error(f"Failed to create database: {e}")
             raise PostgreSQLCreateDatabaseError() from e
-
-        # Enable preset extensions
-        self.enable_disable_extensions(dict.fromkeys(plugins, True), database)
+        finally:
+            cursor.close()
+            connection.close()
 
     def create_user(
         self,
         user: str,
         password: Optional[str] = None,
-        admin: bool = False,
-        extra_user_roles: Optional[List[str]] = None,
+        roles: Optional[List[str]] = None,
+        database: Optional[str] = None,
     ) -> None:
         """Creates a database user.
 
         Args:
             user: user to be created.
             password: password to be assigned to the user.
-            admin: whether the user should have additional admin privileges.
-            extra_user_roles: additional privileges and/or roles to be assigned to the user.
+            roles: roles to be assigned to the user.
         """
-        try:
-            # Separate roles and privileges from the provided extra user roles.
-            admin_role = False
-            roles = privileges = None
-            if extra_user_roles:
-                admin_role = PERMISSIONS_GROUP_ADMIN in extra_user_roles
-                valid_privileges, valid_roles = self.list_valid_privileges_and_roles()
-                roles = [
-                    role
-                    for role in extra_user_roles
-                    if role in valid_roles and role != PERMISSIONS_GROUP_ADMIN
-                ]
-                privileges = {
-                    extra_user_role
-                    for extra_user_role in extra_user_roles
-                    if extra_user_role not in roles and extra_user_role != PERMISSIONS_GROUP_ADMIN
-                }
-                invalid_privileges = [
-                    privilege for privilege in privileges if privilege not in valid_privileges
-                ]
-                if len(invalid_privileges) > 0:
-                    logger.error(f"Invalid extra user roles: {', '.join(privileges)}")
-                    raise PostgreSQLCreateUserError(INVALID_EXTRA_USER_ROLE_BLOCKING_MESSAGE)
 
+        createdb_enabled, createrole_enabled = False, False
+        if "pgbouncer" in roles:
+            createdb_enabled, createrole_enabled = True, True
+            roles.remove("pgbouncer")
+
+        if "admin" in roles:
+            createdb_enabled = True
+            roles.append(ROLE_DML)
+            roles.append(ROLE_INSTANCE_ADMIN)
+            roles.remove("admin")
+
+        if "createdb" in roles:
+            createdb_enabled = True
+            roles.remove("createdb")
+
+        if "createrole" in roles:
+            createrole_enabled = True
+            roles.remove("createrole")
+
+        if "superuser" in roles:
+            logger.warning("superuser privileges not allowed via extra-user-roles")
+            roles.remove("superuser")
```

Comment:
Won't this break PgBouncer? I believe PgBouncer needs the `SUPERUSER` role to grant the `SUPERUSER` role to Landscape or another application when it connects to the PgBouncer legacy relation called `db-admin`. I know that it might not apply to this charm version (PG 16), but we may need to handle this on PgBouncer side, to avoid it breaking.

Reply:
The PGB auth user also can't be created, since it needs access to user password hashes.

Reply:
I was wondering if in PG 16 only, we can translate `SUPERUSER` extra-user-role to `CREATEDB,CREATEROLE` in postgresql charm. if pgbouncer does not need superuser, it can continue to request superuser, and we can just provide it with what it needs in PG 16. we will also add support for `pgbouncer` extra-user-role which pgbouncer can request once we build in support for sharing PG versions through data interfaces